lib/widgets/package_list_view.dart

```diff
@@ -1,25 +1,60 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_pagewise/flutter_pagewise.dart';
 import 'package:pub_client/pub_client.dart';
+import 'package:tavern/screens/bloc.dart';
 import 'package:tavern/widgets/package_tile.dart';
 
-class PackageListView extends StatelessWidget {
-  final Page page;
+class PackageListView extends StatefulWidget {
+  final PageQuery pageQuery;
 
-  const PackageListView({Key key, @required this.page})
-      : assert(page != null),
+  const PackageListView({Key key, @required this.pageQuery})
+      : assert(pageQuery != null),
         super(key: key);
+
+  @override
+  _PackageListViewState createState() => _PackageListViewState();
+}
+
+class _PackageListViewState extends State<PackageListView> {
+  Key pageWiseSliverListKey = GlobalKey();
+
   @override
   Widget build(BuildContext context) {
-    return SliverList(
-      delegate: SliverChildBuilderDelegate(
-        (context, index) {
-          return PackageTile(
-            page: page,
-            index: index,
-          );
-        },
-        childCount: page.packages.length,
-      ),
+    return PagewiseSliverList<Package>(
```

Comment:
you can refactor this to not need a Future and just use BlocBuilder to re-render when a new state is received

Reply:
I don't understand how when the widget requires a future. I'd have to fork and restructure the custom widget to do so